queries/gdscript/highlights.scm

```diff
@@ -41,16 +47,46 @@
   (name) @function (parameters
     (identifier) @parameter)*)
 
-(typed_parameter
-  (identifier) @parameter)
+(typed_parameter (identifier) @parameter)
+(default_parameter (identifier) @parameter)
+
+(call (identifier) @function.call)
+(call (identifier) @include
+      (#any-of? @include "preload" "load"))
+
+;; Properties and Methods
+
+; We'll use @property since that's the term Godot uses.
+; But, should (source (variable_statement (name))) be @property, too? Since a
+; script file is a class in gdscript.
+(class_definition
+  (body (variable_statement (name) @property)))
+
+; Same question but for methods?
+(class_definition
+  (body (function_definition (name) @method)))
 
-(default_parameter
-  (identifier) @parameter)
+(attribute_call (identifier) @method.call)
+(attribute (_) (identifier) @property)
 
-(call
-  (identifier) @function)
-(attribute_call
-  (identifier) @function)
+;; Enums
+
+(enumerator left: (identifier) @constant)
+
+;; Special Builtins
+
+((identifier) @variable.builtin
+  (#any-of? @variable.builtin "self" "super"))
+
+((attribute (identifier) (attribute_call (identifier) @constructor))
+ (#eq? @constructor "new"))
```

Comment:
constructor is not for the keyword new but for the type identifier to be constructed. C++ uses `@keyword.operator` for `new`.

Reply:
Done. Also realized (attribute (identifier)) still matches all attribute calls so removed that nesting.